dapr-1.15.yaml

```diff
@@ -0,0 +1,238 @@
+          tags: $GO_TAGS
+
+test:
+  environment:
+    contents:
+      packages:
+        - kubectl
+        - git
+        - curl
+        - jq
+        - python3
+        - dapr
+        - dapr-daprd
+        - dapr-injector
+        - dapr-placement
+        - dapr-operator
+        - dapr-scheduler
+        - dapr-sentry
+    environment:
+      NAMESPACE: "default"
+  pipeline:
+    - uses: test/kwok/cluster
+    - name: Install stub daprsystem config
+      runs: |
+        kubectl create configmap daprsystem -n default || true
+    - name: Install Dapr CRDs
+      runs: |
+        kubectl wait --for=condition=Ready nodes --all
+        curl -fsSL https://raw.githubusercontent.com/dapr/dapr/v${{package.version}}/charts/dapr/crds/components.yaml | kubectl apply -f -
+        # Give it a sec to settle
+        sleep 2
+    - name: Launch Dapr Operator and verify component processing
+      uses: test/daemon-check-output
+      with:
+        start: operator
+        expected_output: "Starting Dapr Operator"
+        post: "echo \"Deploying test Component...\"\ncat <<EOF | kubectl apply --validate=false -f -\napiVersion: dapr.io/v1alpha1\nkind: Component\nmetadata:\n  name: test-comp\n  namespace: default\nspec:\n  type: bindings.cron\n  version: v1\n  metadata:\n    - name: schedule\n      value: \"@every 5s\"\n    - name: direction\n      value: input\nEOF\n\necho \"⏳ Waiting for component to be processed...\"\nsleep 5\n\necho \"\U0001F50D Verifying component...\"\nkubectl get component test-comp -o yaml | grep -q \"bindings.cron\" \\\n  && echo \"✅ Operator processed component\" \\\n  || (echo \"❌ Component not processed\" && exit 1)\n"
+    - name: Prepare Dapr configs
+      runs: |
+        mkdir -p /etc/dapr
+        echo '{"id": "testapp", "appPort": 8080}' > /etc/dapr/config.json
+        cat <<EOF > /etc/dapr/component.yaml
+        apiVersion: dapr.io/v1alpha1
+        kind: Component
+        metadata:
+          name: testpubsub
+        spec:
+          type: pubsub.in-memory
+          version: v1
+          metadata: []
+        EOF
+    - name: Start placement service
+      uses: test/daemon-check-output
+      with:
+        start: placement
+        expected_output: "placement server leadership acquired"
+        timeout: 15
+    - name: Start sentry with fake config
+      uses: test/daemon-check-output
+      with:
+        start: sentry --config /etc/dapr/daprsystem
+        expected_output: "Starting Dapr Sentry certificate authority"
+        timeout: 10
+    - name: Start scheduler
+      uses: test/daemon-check-output
+      with:
+        start: scheduler
+        expected_output: "Dapr Scheduler listening on: :50006"
+        timeout: 10
+    # We can only check that it starts and fails in melange
+    - name: Start injector (expect failure without env)
+      uses: test/daemon-check-output
+      with:
+        start: injector
+        expected_output: "Error getting config"
+        timeout: 10
```

Comment:
Should we have a `post:` step on each one to await to port and simple `curl | grep` test, if possible?

Reply:
not everyone has a port so I have added for the ones that needed them, thanks for this brother